integration/apps/pennant/README.md

```diff
@@ -1,5 +1,35 @@
+# What is Pennant?
+
 From https://asc.llnl.gov/coral-2-benchmarks:
 
     PENNANT is a mini-app for hydrodynamics on general unstructured
     meshes in 2D (arbitrary polygons). It makes heavy use of indirect
-    addressing and irregular memory access patterns.
\ No newline at end of file
+    addressing and irregular memory access patterns.
+
+A detailed description of the algorithm can be found in the Pennant
+repository (downloaded to integration/apps/pennant/PENNANT by the
+build.sh script) under doc/pennantdoc.pdf.
+
+# Number of Cores Reserved for Pennant
+
+Pennant AppConfig file (pennant.py) reserves N physical cores per node
+where N is the largest even number less than the total number of physical
+cores in the node. This way, at least one physical core is reserved for
+the OS and GEOPM.
+
+# GEOPM Epoch Selection
+
+Based on the problem size we choose a different build of pennant.
+The difference between these builds is how GEOPM epochs are marked:
+
+    epoch100: One epoch every 100 outer loops iterations.
+    epoch1: One epoch every 100 outer loops iterations.
```

Comment:
what's the difference between these?

Reply:
That was a typo. Fixed it.
    epoch100: One epoch every 100 outer loop iterations.
    epoch1: One epoch every outer loop iteration.